src/parser/rogue/assassination/modules/talents/Nightstalker.js

```diff
@@ -25,28 +26,26 @@ class Nightstalker extends Analyzer {
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTalent(SPELLS.NIGHTSTALKER_TALENT.id);
-  }
-
-  on_byPlayer_damage(event) {
-    if (!this.selectedCombatant.hasBuff(SPELLS.STEALTH.id)) {
+    if (!this.active) {
       return;
     }
-    const spellId = event.ability.guid;
-    if (!ABILITIES_AFFECTED_BY_DAMAGE_INCREASES.includes(spellId)) {
-      return;
-    }
-    if (ABILITIES_AFFECTED_BY_SNAPSHOTTING.includes(spellId)) {
+    ABILITIES_AFFECTED_BY_DAMAGE_INCREASES.forEach(e => {
```

Comment:
I'm not sure if comparing objects directly would work. Might be more elegant to do this:
```javascript
const allowedAbilities = ABILITIES_AFFECTED_BY_DAMAGE_INCREASES.filter(spell => !NIGHTSTALKER_BLACKLIST.some(blacklisted => blacklisted.id === spell.id));
this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(allowedAbilities), this.addBonusDamageIfBuffed);
```
`some()` returns true if it finds any item satisfying the condition. So if it doesn't find the spell in the BLACKLIST with the same ID, it's allowed.

Reply:
It works, they point to the same object. Anyway your solution is prettier, so I'll use that instead.

Reply:
I mean, if it works, you might as well simplify it even more with include
```javascript
const allowedAbilities = ABILITIES_AFFECTED_BY_DAMAGE_INCREASES.filter(spell => !NIGHTSTALKER_BLACKLIST.includes(spell));
```